pkg/controllers/workapplier/status.go

```diff
@@ -268,7 +272,13 @@ func setManifestAppliedCondition(
 		}
 	}
 
-	meta.SetStatusCondition(&manifestCond.Conditions, *appliedCond)
+	if appliedCond != nil {
+		meta.SetStatusCondition(&manifestCond.Conditions, *appliedCond)
+	} else {
+		// As the conditions are ported back; removal must be performed if the DiffReported
+		// condition is not set.
+		meta.RemoveStatusCondition(&manifestCond.Conditions, fleetv1beta1.WorkConditionTypeApplied)
+	}
```

Comment:
I don't see any test changes that is related to most of the changes in this file

Reply:
Will add more tests. Before there was the ReportDiff mode UT that covers this case (`Applied` -> `DiffReported`, no left over conditions); can always use more of course.